auth/auth.go

```diff
@@ -0,0 +1,7 @@
+package auth
+
+import "net/http"
+
+type Authenticator interface {
```

Comment:
is this a public user facing API?

Reply:
It is a public facing interface. Only needed if user wants to have their own implementation.

Reply:
@andrefurlan-db I am planning to write the decision doc we discussed sometime end of this week / early next week. as this is a change in the public surface area, can we wait to close on that decision before merging this?

Reply:
This PR is target for version 1.1.0. It won't be merged this year.